pkg/controller/node/ipam/cloud_cidr_allocator.go

```diff
@@ -191,10 +191,14 @@ func (ca *cloudCIDRAllocator) AllocateOrOccupyCIDR(node *v1.Node) error {
 
 // updateCIDRAllocation assigns CIDR to Node and sends an update to the API server.
 func (ca *cloudCIDRAllocator) updateCIDRAllocation(nodeName string) error {
-	var err error
-	var node *v1.Node
 	defer ca.removeNodeFromProcessing(nodeName)
 
+	node, err := ca.nodeLister.Get(nodeName)
+	if err != nil {
+		glog.Errorf("Failed while getting node %v for updating Node.Spec.PodCIDR: %v", nodeName, err)
+		return err
+	}
+
```

Comment:
I'm not familiar to the code here, but it seems that L218-221 is related and probably should be removed?

Reply:
I just removed `approved`, and I'll add it back once this is resolved.

Reply:
yes -- looks like the logging below is redundant.

Reply:
@jingax10 commented earlier that the 2nd call is needed to refresh the podcidr state on retries.